### PR TITLE
Princejha/removeobjectid

### PR DESCRIPTION
**Pull request by princejha95**
Changes to remove passing of sp object id while k8 extension creation and removing hardcoding of azure cli version by changing it to always use the latest one.

### CODE REVIEW

File: src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py [hunk 1]
```diff
@@ -142,7 +140,7 @@ def _check_if_required_features_registered(self):
             required_features_list = item['feature']
             namespace: str = item['namespace']
             if type(required_features_list) != list:
-                required_features_list = [required_features_list,]
+                required_features_list = [required_features_list, ]
```
Comment (msfttusharjuneja) [inline src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py:143]: Is this required?

**Comment by msfttusharjuneja** [comment]
Which utility was used for code refactoring?

File: src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py (hunk 1 above)
Comment (princejha95) [reply src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py]: Fixed it

**Comment by princejha95** [comment]
> Did you test the Linux script? There are changes related to vmwareSPObjectID in the Linux version of the script and if we don't change that, I suspect it might break the shell script.

Tested both Onboarding and offboarding in linux JB as well

**Comment by princejha95** [comment]
> Which utility was used for code refactoring?

Pep8. Comes inbuilt with Pycharm

**Comment by msfttusharjuneja** [comment]
> 

There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.

**Comment by princejha95** [comment]
> > 
> 
> There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.

There is no such variable in run.sh file

**Comment by msfttusharjuneja** [comment]
> > 
> 
> There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.

https://github.com/Azure/ArcOnAVS/blob/5a9bf8e6ea916995cf6feecb2ccd65e484fdbe73/src/appliance-onboarding-script/run.sh#L133

**Comment by princejha95** [comment]
> > 
> 
> There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.

I will remove it but why onboarding is not failing if i did not removed it while testing ?

**Comment by msfttusharjuneja** [comment]
> > > 
> > 
> > 
> > There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.
> 
> I will remove it but why onboarding is not failing if i did not removed it while testing ?

My understanding is that '' (empty string) value gets passed since no value was given for $5. The handling in the python file is also removed, hence this param gets ignored. This can be a problem in the future when we introduce new parameters and hence should be removed. Please feel free to investigate this further as I might be incorrect

**Comment by princejha95** [comment]
> > > > 
> > > 
> > > 
> > > There is $5 variable in the linux script meant for vmwarespobjectid, which should be removed.
> > 
> > 
> > I will remove it but why onboarding is not failing if i did not removed it while testing ?
> 
> My understanding is that '' (empty string) value gets passed since no value was given for $5. The handling in the python file is also removed, hence this param gets ignored. This can be a problem in the future when we introduce new parameters and hence should be removed. Please feel free to investigate this further as I might be incorrect

done

**Review by uthsab — APPROVED** [review]
Changes look good.  Added some NIT comments

File: src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py [hunk 2]
```diff
@@ -142,7 +140,7 @@ def _check_if_required_features_registered(self):
             required_features_list = item['feature']
             namespace: str = item['namespace']
             if type(required_features_list) != list:
-                required_features_list = [required_features_list,]
+                required_features_list = [required_features_list]
```
[the remark this answers is not in the document]
Comment (princejha95) [reply src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py:143]: Yes

File: src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py [hunk 3]
```diff
@@ -284,36 +283,25 @@ def _create_or_delete_vmware_extension(self, op='create') -> str:
         except KeyError:
             pass
 
-        if self._input_vmware_sp_object_id is None:
```
[the remark this answers is not in the document]
Comment (princejha95) [reply src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py:287]: Yes, right. Will address it in next PR

**Pull request merged**